src/ray/raylet/node_manager.cc

```diff
@@ -1865,6 +1869,49 @@ void NodeManager::HandleRequestWorkerLease(rpc::RequestWorkerLeaseRequest reques
                                               send_reply_callback_wrapper);
 }
 
+void NodeManager::HandlePrestartWorkers(rpc::PrestartWorkersRequest request,
+                                        rpc::PrestartWorkersReply *reply,
+                                        rpc::SendReplyCallback send_reply_callback) {
+  const uint64_t num_workers = std::min(
+      RayConfig::instance().restart_workers_api_max_num_workers(), request.num_workers());
+  if (num_workers < request.num_workers()) {
+    RAY_LOG(WARNING) << "Requested to prestart " << request.num_workers()
+                     << " workers, but only " << num_workers
+                     << " workers are allowed to prestart. See "
+                        "RAY_restart_workers_api_max_num_workers";
+  }
+
+  auto pop_worker_request = std::make_shared<PopWorkerRequest>(
```

Comment:
ok

Reply:
`StartNewWorker(PopWorkerRequest)` is also used by `PopWorker(PopWorkerRequest)` internally. The Request does not have anything private so I guess we can just expose it and allow node_manager.cc to use it? It will be much easier. If we just expose another method with the scattered 10 arguments it doesn't have any added values any way.